editor/code_editor.cpp

```diff
@@ -1956,11 +1908,49 @@ void CodeTextEditor::remove_all_bookmarks() {
 	text_editor->clear_bookmarked_lines();
 }
 
+void CodeTextEditor::zoom_in() {
+	int s = text_editor->get_theme_font_size("font_size");
+	set_zoom_factor(zoom_factor * (s + MAX(1.0f, EDSCALE)) / s);
+}
+
+void CodeTextEditor::zoom_out() {
+	int s = text_editor->get_theme_font_size("font_size");
+	set_zoom_factor(zoom_factor * (s - MAX(1.0f, EDSCALE)) / s);
+}
+
+void CodeTextEditor::set_zoom_factor(float p_zoom_factor) {
+	if (zoom_factor == p_zoom_factor) {
+		return;
+	}
+
+	int preset_count = sizeof(ZOOM_FACTOR_PRESETS) / sizeof(float);
+	float new_zoom_factor = CLAMP(p_zoom_factor, ZOOM_FACTOR_PRESETS[0], ZOOM_FACTOR_PRESETS[preset_count - 1]);
+	int neutral_font_size = EDITOR_GET("interface/editor/code_font_size").operator int() * EDSCALE;
```

Comment:
Use simpler casting form:

```suggestion
	int neutral_font_size = int(EDITOR_GET("interface/editor/code_font_size")) EDSCALE;
```